optuna/integration/sklearn.py

```diff
@@ -69,6 +68,22 @@ def _check_sklearn_availability():
         )
 
 
+def _is_arraylike(x):
+    # type: (Any) -> bool
+    return hasattr(x, '__len__')
```

Comment:
https://github.com/scikit-learn/scikit-learn/blob/c2ede74db053d4821c51e1de53f9f8522562e9b5/sklearn/utils/validation.py#L131
Should we also check `__array__` and `shape`?